unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/StickersController/StickersController.cs

```diff
@@ -30,21 +38,49 @@ public void PlaySticker(string id, Vector3 position, Vector3 direction, bool fol
             if (DCL.Configuration.EnvironmentSettings.RUNNING_TESTS)
                 return;
 
-            GameObject emoteGameObject = Instantiate(prefab);
-            emoteGameObject.transform.position += position;
-            emoteGameObject.transform.rotation = Quaternion.Euler(prefab.transform.rotation.eulerAngles + direction);
+            PoolableObject sticker = pools[id].Get();
+            GameObject stickerGo = sticker.gameObject;
+            stickerGo.transform.position += position;
+            stickerGo.transform.rotation = Quaternion.Euler(prefab.transform.rotation.eulerAngles + direction);
+
+            ReleaseParticlesOnFinish releaser = stickerGo.GetComponent<ReleaseParticlesOnFinish>();
+            if (releaser != null)
+                releaser.Initialize(sticker);
 
             if (followTransform)
             {
-                FollowObject emoteFollow = emoteGameObject.AddComponent<FollowObject>();
-                emoteFollow.target = transform;
-                emoteFollow.offset = position;
+                FollowObject stickerFollow = stickerGo.AddComponent<FollowObject>();
```

Comment:
Can this cause `FollowObject.cs` to be added multiple times if the object is pooled?
If yet, then we can create and extension method, i.e `public static T GetOrAddComponent<T>(this GameObject gameObject)`

Reply:
Fantastic catch!
This was not happening before because the object was destroyed, but now that we are pooling it this would bbecome an issue, correcting it

Reply:
Nice catch Aga! we have that extension method already.